test/d2l-filter-dropdown/d2l-filter-dropdown.js

```diff
@@ -79,289 +79,317 @@ import { afterNextRender } from '@polymer/polymer/lib/utils/render-status.js';
 	}
 
 	suite('d2l-filter-dropdown', function() {
-		setup(function(done) {
+
+		setup(() => {
 			filter = fixture('basic');
-			categories = [
-				{key: 'cat1', title: 'Cat 1'},
-				{key: 'cat2', title: 'Cat 2'},
-				{key: 'cat3', title: 'Cat 3'}
-			];
-			options = [
-				{cat: 'cat1', key: 'opt1-1', title: 'Opt 1-1'},
-				{cat: 'cat1', key: 'opt1-2', title: 'Opt 1-2'},
-				{cat: 'cat2', key: 'opt2-1', title: 'Opt 2-1'},
-				{cat: 'cat2', key: 'opt2-2', title: 'Opt 2-2'},
-				{cat: 'cat3', key: 'opt3-1', title: 'Opt 3-1'},
-				{cat: 'cat3', key: 'opt3-2', title: 'Opt 3-2'}
-			];
-			afterNextRender(filter, done);
-		});
-		test('instantiating the element works', function() {
-			assert.equal('d2l-filter-dropdown', filter.tagName.toLowerCase());
-		});
-		test('attributes are set correctly', function() {
-			assert.equal(false, filter.disableSearch);
-		});
-		test('filters are imported correctly', function() {
-			var expected = _getExpectedAndImport(filter);
-			assert.deepEqual(expected, filter._filters);
 		});
-		test('clear button is hidden if no filters are applied', function() {
-			var clearButton = filter.shadowRoot.querySelector('d2l-button-subtle');
-			assert.isTrue(clearButton.hidden);
-		});
-		test('filter category num selected is correct after importing filters and clear button is visible', function() {
-			var clearButton = filter.shadowRoot.querySelector('d2l-button-subtle');
-			var selected = [0, 4, 5];
-			var expectedNum = _setSelectedOptions(selected);
-			var expected = _getExpectedAndImport(filter);
 
-			assert.isFalse(clearButton.hidden);
-			assert.deepEqual(expected, filter._filters);
-			for (var i = 0; i < filter._filters.length; i++) {
-				var f = filter._filters[i];
-				for (var j = 0; j < f.options.length; j++) {
-					assert.equal(f.options[j].selected, _getOptionByKeys(f.key, f.options[j].key).selected);
-				}
-				assert.equal(expectedNum[f.key], f.numSelected);
-			}
+		suite('unit', () => {
```

Comment:
I tried to write integration tests for this function originally but I couldn't figure it out. However, I'm confident that these unit tests prove the function does what it's supposed to.